### DISCUSSION TITLE
select one of three servo motor position when object is visible

### DISCUSSION THREAD

**Discussion by salvatoreNRIM** [bonsai-rx/bonsai#1092] [Q&A]
Hi there,

I want to change the position of a servo motor (either turn it to the left 90 deg, leave it in the same position 100 deg, or turn it to right 110 deg) once the mouse is visible in the camera, this position will be maintained for 5 seconds and then put it back to 100 deg until the next time the mouse is visible again. I tried to use the permutation function, but I get some errors (below a screenshot of the bonsai script), and the motor never changes position. Would anyone have suggestions on this?

![image](https://user-images.githubusercontent.com/46469676/196360328-2c845031-b577-4a8f-b3c6-12a723725277.png)

Best,
Salvo

**Comment by bruno-f-cruz** [comment]
Hey,

The first error is due to the datatype you are trying to pass to the `ServoOutput` node. It expects an `Int` and you are inputting a `Tuple<int, bool>`, try to only select the first element of the Tuple.

Regarding the permutation, check the attached workflow for two possible solutions.

Cheers,
B
[PermutationExample.zip](https://github.com/bonsai-rx/bonsai/files/9807633/PermutationExample.zip)

**Comment by salvatoreNRIM** [comment]
Hi Bruno,

Thanks for the answer! I tried to incorporate your script such that it permutates the platform when the mouse is visible in the camera. But when I try to use the zip node to put together the RandomTilt value and the condition that the mouse is in a certain position, the servo output node does not like it...

[Perm_example.zip](https://github.com/bonsai-rx/bonsai/files/9809480/Perm_example.zip)



![image](https://user-images.githubusercontent.com/46469676/196408322-89b4e17b-7686-46d7-9c36-874bb97ace79.png)

**Reply by bruno-f-cruz** [reply]
Can you try the attached workflow? I am not sure what you are trying to achieve with `Zip`, but from your description, I think this is the behavior you are looking for.
[Select_servo_trigger_closed_loop_perm.zip](https://github.com/bonsai-rx/bonsai/files/9809582/Select_servo_trigger_closed_loop_perm.zip)

**Reply by salvatoreNRIM** [reply]
Hi Bruno! Maybe there was a type in your code (in RandomDraw there should be a take after the first node). After adding that your code was working perfectly!
